client/src/components/DisplayTable/DisplayTable.js

```diff
@@ -113,6 +122,37 @@ export class DisplayTable extends React.Component {
     }
   }
 
+  change_page_size = (new_page_size) =>
+    this.state.page_size !== new_page_size &&
+    this.setState({
+      show_pagination_load_spinner: true,
+      page_size: new_page_size,
+      current_page: 0,
+    });
+
+  change_page = (page) =>
+    this.state.current_page !== page &&
+    this.setState({
+      show_pagination_load_spinner: true,
+      current_page: page,
+    });
+
+  debounced_stop_loading = _.debounce(
+    () => this.setState({ show_pagination_load_spinner: false }),
+    1
+  );
+
+  componentDidUpdate(prev_props, prev_state) {
+    const { show_pagination_load_spinner } = this.state;
+    if (show_pagination_load_spinner) {
+      this.debounced_stop_loading();
```

Comment:
Debounce is still the wrong tool here, it's the time out that's relevant, right?

Add a comment in `componentDidUpdate` on why we're using this hackyish pattern (both in general why we need a loading spinner on pagination change and in particular why it needs to be bumped back in the event queue with a 1ms timeout).